subsys/net/l2/ethernet/arp.c

```diff
@@ -608,10 +631,13 @@ enum net_verdict net_arp_input(struct net_pkt *pkt,
 				   &arp_hdr->src_ipaddr,
 				   &arp_hdr->src_hwaddr,
 				   false, true);
+
+			dst_hw_addr = &arp_hdr->src_hwaddr;
 		}
```

Comment:
add an: 
else {
    dst_hw_addr = &eth_hdr->src.addr;
}

It is then more obvious what is dst_addr value and why.